service/config_provider.go

```diff
@@ -154,5 +160,44 @@ func (cm *configProvider) closeIfNeeded(ctx context.Context) error {
 
 func (cm *configProvider) Shutdown(ctx context.Context) error {
 	close(cm.watcher)
-	return multierr.Combine(cm.closeIfNeeded(ctx), cm.configMapProvider.Shutdown(ctx))
+	return multierr.Combine(cm.closeIfNeeded(ctx), mergeShutdown(ctx, cm.configMapProviders))
+}
+
+func mergeRetrieve(ctx context.Context, onChange func(*configmapprovider.ChangeEvent), providers []configmapprovider.Provider) (configmapprovider.Retrieved, error) {
+	var retrs []configmapprovider.Retrieved
+	retCfgMap := config.NewMap()
+	for _, p := range providers {
+		retr, err := p.Retrieve(ctx, onChange)
+		if err != nil {
+			return nil, err
+		}
+		cfgMap, err := retr.Get(ctx)
```

Comment:
Is there ever a reason to call `Retrieve()` independent of `Get()`, or vice versa?  Does this need to be a two-stage process?

Reply:
Consider to read the API:
https://github.com/open-telemetry/opentelemetry-collector/blob/main/config/configmapprovider/provider.go#L38
https://github.com/open-telemetry/opentelemetry-collector/blob/main/config/configmapprovider/retrieved.go#L26

In summary the decision to retrieve an object with two function was preferred via returning the map + a close function. If you think the other option is better please file a short draft PR with how you would envision that, or an issue.

Reply:
Please "unresolve" this comment if you feel the answer is not good enough.